lib/seeing_is_believing/the_matrix.rb

```diff
@@ -1,11 +1,21 @@
 require_relative 'safe'
 require_relative 'version'
 require_relative 'event_stream/producer'
+require 'socket'
+require 'timeout'
 
 using SeeingIsBelieving::Safe
 
 sib_vars     = Marshal.load ENV["SIB_VARIABLES.MARSHAL.B64"].unpack('m0').first
-event_stream = IO.open sib_vars.fetch(:event_stream_fd), "w"
+event_stream = Timeout.timeout(1) do
+  begin
+    Socket.tcp("localhost", sib_vars.fetch(:event_stream_port))
+  rescue Errno::ECONNREFUSED
+    sleep 0.1
+    retry
+  end
+end
```

Comment:
I'm not thrilled with this little kludge. That said accepting the connection in a thread would not eliminate the race condition. There still no guarantee that the thread would be ready by the time the child process got up to speed.